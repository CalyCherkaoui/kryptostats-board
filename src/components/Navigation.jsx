/* eslint-disable no-console */
import React from 'react';
import { Link } from 'react-router-dom';
import SearchBar from './SearchBar';
// import LocalCurrencyFilter from './LocalCurrencyFilter';

function Navigation() {
  return (
    <div className="navigation_bar">
      <div className="navigation_left">
        <div>
          <h1 className="nav_logo">Krypto-stats | Dashboard</h1>
          <ul>
            <li className="nav_item">
              <Link to="/about" className="nav_link">About</Link>
            </li>
            <li className="nav_item">
              <Link to="/" className="nav_link">Index</Link>
            </li>
          </ul>
        </div>
      </div>
      <div className="navigation_right">
        {/* <LocalCurrencyFilter changeLocalCurrency={(val) => console.log(val)} /> */}
        <SearchBar />
      </div>
    </div>
  );
}

export default Navigation;
