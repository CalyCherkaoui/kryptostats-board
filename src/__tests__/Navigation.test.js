import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Navigation from '../components/Navigation';

describe(
  'Renders correctly The navigation bar component',
  () => {
    let container;
    beforeEach(() => {
      container = shallow(
        <Provider store={store}>
          <Router>
            <Navigation />
          </Router>
        </Provider>,
      );
    });

    it(
      'Contains SearchBar component',
      () => {
        expect(container.find('SearchBar')).toBeTruthy();
      },
    );

    it(
      'Matches Navigation snapshot',
      () => {
        const rendered = renderer.create(
          <Provider store={store}>
            <Router>
              <Navigation />
            </Router>
          </Provider>,
        );

        expect(rendered).toMatchSnapshot();
      },
    );
  },
);
