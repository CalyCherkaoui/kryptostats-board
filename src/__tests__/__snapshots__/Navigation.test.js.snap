// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly The navigation bar component Matches Navigation snapshot 1`] = `
<div
  className="navigation_bar flex_row flex_space_between shadowed_small"
>
  <div
    className="navigation_left flex_row flex_start_bl"
  >
    <p
      className="nav_logo bold_typography"
    >
      Krypto|
      <span
        className="nav_logo_emph"
      >
        <svg
          className="logo_icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="8"
            cy="8"
            r="8"
          />
        </svg>
      </span>
    </p>
    <ul
      className="flex_row flex_start_bl"
    >
      <li
        className="nav_item"
      >
        <a
          className="nav_link spaced_typography"
          href="/about"
          onClick={[Function]}
        >
          ABOUT
        </a>
      </li>
      <li
        className="nav_item"
      >
        <a
          className="nav_link spaced_typography"
          href="/"
          onClick={[Function]}
        >
          CRYPTO-LIST
        </a>
      </li>
    </ul>
  </div>
  <div
    className="navigation_right"
  >
    <div
      className="search_bar_container"
    >
      <input
        className="search_bar_input normal_typography"
        onChange={[Function]}
        placeholder="Search Crypto"
        type="text"
      />
      <button
        className="search_bar_button"
        onClick={[Function]}
        type="button"
      >
        Search
      </button>
    </div>
  </div>
</div>
`;
